src/com/goide/GoDocumentationProvider.java

```diff
@@ -102,6 +104,64 @@ else if (type == GoParserDefinition.MULTILINE_COMMENT) {
         }
         return text;
       }
-    }), "<br/>") + "</pre>";
+    }), "<br/>");
+  }
+
+  @NotNull
+  private static String getSignature(PsiElement element) {
+    if (!(element instanceof GoSignatureOwner)) {
+      return "";
+    }
+
+    PsiElement identifier = null;
+    if (element instanceof GoNamedSignatureOwner) {
+      identifier = ((GoNamedSignatureOwner)element).getIdentifier();
+    }
+    GoSignature signature = ((GoSignatureOwner)element).getSignature();
+
+    if (identifier == null && signature == null) {
+      return "";
+    }
+
+    String result = " <b>func " + (identifier != null ? identifier.getText() : "") + "(";
```

Comment:
I've took the decision to not display the receiver of the function as I don't think it's relevant for the user to know that information. I've tested in practice both with and without it before committing it like this.